api/query/search.go

```diff
@@ -99,68 +103,97 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 			val, _ := shared.ParseBooleanParam(q, param)
 			isSimpleQ = isSimpleQ && (val == nil || !*val)
 		}
-	}
-
-	if !isSimpleQ {
-		ctx := sh.api.Context()
-		hostname := sh.api.GetServiceHostname("searchcache")
-		// TODO: This will not work when hostname is localhost (http scheme needed).
-		fwdURL, _ := url.Parse(fmt.Sprintf("https://%s/api/search/cache", hostname))
-		fwdURL.RawQuery = r.URL.RawQuery
 
-		logger := shared.GetLogger(ctx)
-		logger.Infof("Forwarding structured search request to %s: %s", hostname, string(data))
+		q.Add("q", url.QueryEscape(simpleQ.Pattern))
+		// Check old summary files. If any can't be found,
+		// use the searchcache to aggregate the runs.
+		for _, id := range rq.RunIDs {
+			q.Add("run_id", strconv.FormatInt(id, 10))
+		}
 
-		client := sh.api.GetHTTPClientWithTimeout(time.Second * 15)
-		req, err := http.NewRequest("POST", fwdURL.String(), bytes.NewBuffer(data))
+		r2 := r.Clone(r.Context())
+		r2.Method = "GET"
+		r2.URL.RawQuery = q.Encode()
+		summariesValid, err := sh.validateSummaryVersions(r2.URL.Query())
 		if err != nil {
-			logger.Errorf("Failed to create request to POST %s: %v", fwdURL.String(), err)
-			http.Error(w, "Error connecting to search API cache", http.StatusInternalServerError)
-			return
+			logger.Debugf("Error checking version files: %v", err)
```

Comment:
If the file format is invalid, we should return `StatusInternalServerError` here

Reply:
I didn't do this just in case there's a corner case in which the summary files are not available for some reason but the search cache can still be used. Also, this was a poor error wording on my part from an old approach and I've changed it to be more clear.